docs/public/components/overview.md

```diff
@@ -0,0 +1,45 @@
+# Overview of components
+
+The Agent is structured as a collection of components working together. Depending on how the binary is built, and how it
+is invoked, different components may be instantiated.
```

Comment:
> Depending on how the binary is built, and how it
is invoked, different components may be instantiated.

I think this sentence is quite misleading. The components created don't depend on the build tags neither on the command line arguments